File: flix_backend/lib/flix_backend/data/school.ex
defmodule FlixBackend.Data.School do
  use Ecto.Schema
  import Ecto.Changeset

  @primary_key {:id, :binary_id, autogenerate: true}
  @derive {Phoenix.Param, key: :id}
  @derive {Jason.Encoder, only: [:id, :name, :code]}
  schema "schools" do
    field :name, :string
    field :code, :string

    # 反向引用校区
    has_many :campuses, FlixBackend.Data.Campus, foreign_key: :school_id

    timestamps()
  end

  def changeset(school, attrs) do
    school
    |> cast(attrs, [:name, :code])
    |> validate_required([:name])
    |> unique_constraint(:name)
  end
end

File: Flix/app/src/main/java/fin/phoenix/flix/data/School.kt
package fin.phoenix.flix.data

import com.google.gson.annotations.SerializedName

/**
 * 学校数据类
 */
data class School(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("address") val address: String? = null,
    @SerializedName("city") val city: String? = null
)

/**
 * 校区数据类
 */
data class Campus(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("school_id") val schoolId: String,
    @SerializedName("address") val address: String? = null
)